lib/assigner.rb

```diff
@@ -523,7 +523,15 @@ def topic_same_assignee_and_details(assign_to, type, note, status)
   end
 
   def post_same_assignee_and_details(assign_to, type, note, status)
-    @target.is_a?(Topic) &&
+    if @target.is_a?(Post)
+      @target.assignment&.assigned_to_id == assign_to.id &&
+      @target.assignment&.assigned_to_type == type && @target.assignment.active == true &&
+      @target.assignment&.note == note &&
+      (
+        @target.assignment&.status == status ||
+          @target.assignment&.status == Assignment.default_status && status.nil?
```

Comment:
Should this be inside a parenthesis? Otherwise it's hard to know if `||` has precedence over `&&`

```suggestion
          (@target.assignment&.status == Assignment.default_status && status.nil?)
```

Reply:
I agree it can be confusing at first. I changed the other places too.